src/python/pants/backend/native/tasks/link_shared_libraries.py

```diff
@@ -142,6 +143,16 @@ def _make_link_request(self, vt, compiled_objects_product, native_target_deps_pr
   def _get_shared_lib_cmdline_args(self):
     return self.linker.platform.resolve_platform_specific(self._SHARED_CMDLINE_ARGS)
 
+  def _get_third_party_lib_args(self):
+    lib_args = []
+    tp_files_product = self.context.products.get_data(NativeExternalLibraryFetch.NativeExternalLibraryFiles)
```

Comment:
This product dependency *needs* to be required in a `prepare()` method for this class.

In general I really prefer to fetch all products in the `execute()` method, because it makes it much easier to see what products a task depends on, and where, and I would suggest you do that here (and just pass it in as an argument).

Reply:
:white